openapi.yaml

```diff
@@ -132,147 +462,375 @@ securityDefinitions:
-  PaginationLinks:
+        description: the type of the resource (pizza)
+        enum: [pizza]
+      id: &pizzaId
+        type: string
+        description: a unique ID of a pizza
+        example: abc123
+      links:
+        $ref: "#/definitions/SelfLink"
+      attributes:
+        $ref: "#/definitions/PizzaAttributes"
+      relationships: &pizzaRelationships
+        properties:
+          dough:
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the dough referenced
+                    example: https://api.oregonstate.edu/v1/doughs/1
+              data:
+                $ref: "#/definitions/DoughRelationshipResource"
+          ingredients:
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the ingredient referenced
+                    example: https://api.oregonstate.edu/v1/ingredients/1
+              data:
+                type: array
+                description: a list of relationship resources with the ingredient type
+                items:
+                  $ref: "#/definitions/IngredientRelationshipResource"
+      included: &pizzaIncluded
+        type: array
+        items:
+          type: object
+          description: An ingredient resource or a dough resource
+  PizzaPostBody:
+    required: [data]
+    properties:
+      data:
+        required: [type, attributes]
+        properties:
+          type: *pizzaType
+          attributes:
+            $ref: "#/definitions/PizzaAttributes"
+          relationships: &pizzaUpdateRelationships
+            properties:
+              dough:
+                properties:
+                  data:
+                    $ref: "#/definitions/DoughRelationshipResource"
+              ingredients:
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: "#/definitions/IngredientRelationshipResource"
+  PizzaPatchBody:
+    required: [data]
     properties:
-      first:
+      data:
+        required: [id, type]
+        properties:
+          type: *pizzaType
+          id: *pizzaId
+          attributes:
+            $ref: "#/definitions/PizzaPatchAttributes"
+          relationships: *pizzaUpdateRelationships
+  PizzaAttributes:
+    type: object
+    required: [name, ovenTemp, bakeTime]
```

Comment:
Yeah, this makes sense. I'm going to change this so that `PizzaPatchAttributes` is now `PizzaAttributes`, which has no required fields, and so that `PizzaAttributes`, which has required fields, is now `PizzaPostAttributes`.